scrapper.py

```diff
@@ -44,20 +75,94 @@ def get_search_urls(self):
         pass
 
 
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def _fill_article_with_meta_information(self, article_bs):
+        article_title = article_bs.find('h1').text
+        self.article.title = article_title.strip()
+
+        self.article.author = 'NOT FOUND'
+
+        date_epoch = article_bs.dateformat['time']
+        self.article.date = datetime.datetime.fromtimestamp(int(date_epoch))
+
+        topic = article_bs.find('a', class_='tags__item')
+        if not topic:
+            topic = 'NOT FOUND'
+        else:
+            topic = topic.text
+        self.article.topics = topic
+
+    def _fill_article_with_text(self, article_bs):
+        text_content = article_bs.find('div', class_="text-content")
+        divs = text_content.find_all('div', class_="text-block")
+        self.article.text = ''
+        for div in divs:
+            p_tags = div.find_all('p')
+            for p_tag in p_tags:
+                self.article.text += p_tag.text.strip()
+
+    def parse(self):
+        response = requests.get(url=self.article_url)
```

Comment:
are any headers or specific request settings required? did you tried to parse 100 articles?

Reply:
I added headers, but the code still does not pass the tests on the github (the crawler and parser pass in the PyCharm). And sometimes a connection error occurs((.